File: CSharp/TensionsBetweenDifferentOOPrinciples/TensionBetweenSRPandLowCoupling/After/Plateau.cs
using System;

namespace TensionBetweenSRPandLowCoupling.After
{
    public class Plateau
    {
        // Plateau class represents a plateau with well defined boundaries.
        // So Plateau class follows the Single Responsibility Principle (SRP).

        private readonly Tuple<int, int> _plateau = new Tuple<int, int>(11, 13);

        public bool Contains(Tuple<int, int> position)
        {
            return ((0 <= position.Item1 || position.Item1 <= _plateau.Item1) ||
                    (0 <= position.Item2 || position.Item2 <= _plateau.Item2));
        }
    }
}

File: javascript/TensionsBetweenDifferentOOPrinciples/TensionBetweenSRPandLowCoupling/After/plateau.js
// Plateau class represents a plateau with well defined boundaries.
// So Plateau class follows the Single Responsibility Principle (SRP).
var Plateau = function () {
  this.plateau = [11, 13];
};

Plateau.prototype.contains = function (position) {
  return ((0 <= position[0] && position[0] <= this.plateau[0]) &&
      (0 <= position[1] && position[1] <= this.plateau[1]));
};

module.exports = Plateau;
